### DISCUSSION TITLE
VS Code CodeQL extension displays the results in alerts mode rather than in PATH view format

### DISCUSSION THREAD

**Discussion by Sim4n6** [github/codeql#12491] [Q&A] [answered]
I get this strange behavior in VSCode CodeQL extension. I think I am missing something but I don't know what.

The excerpt of the query is:

```codeql 
/**
 * @name name
 * @description descr 
 * @kind path-problem
 * @id py/uid
 * @security-severity 7.5
 * @precision high
 * @tags security
 *       experimental
 *       external/cwe/cwe-123
 */

import python
//import semmle.python.dataflow.new.DataFlow
import semmle.python.dataflow.new.TaintTracking
import semmle.python.dataflow.new.RemoteFlowSources
import semmle.python.Concepts
import semmle.python.ApiGraphs
import DataFlow::PathGraph
import semmle.python.dataflow.new.internal.DataFlowPublic

class Config extends TaintTracking::Configuration {
  Config() { this = "uid" }
  // override isAdditionalTaintStep + isSanitizer + isSink + isSource (RemoteFlowSource).
}

from Config config, DataFlow::PathNode source, DataFlow::PathNode sink
where config.hasFlowPath(source, sink)
select sink, source, sink, "A potential sweet catch "
```  

I get these results in **alerts** mode, it is supposed to give me results in PATH view format, right? 

![image](https://user-images.githubusercontent.com/13036531/224503260-90c6de05-d06a-497f-9bfb-72fcbf1fb8b3.png)

PS: One last detail ...  please, I sometime comment the query part while leaving the class as-is, and modify the `@kind` to `problem` to test in the same query.  Is it some kind of caching .. . ? 

Thank you in advance.
@Sim4n6

**Comment by smowton** [comment, marked as answer]
It is normal for the view to be named "alerts" even if there are associated paths. It is also normal for the path dropdown not to appear if the path is of length 1 (i.e., usually if the source and sink are in the same function). Could you give example Python and QL that produces an unexpected result?

**Reply by Sim4n6** [reply]
> the source and sink are in the same function 
for that case, indeed, thank you very much
